cmd/attest.go

```diff
@@ -0,0 +1,218 @@
+package cmd
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/proto/attest"
+	"google.golang.org/protobuf/encoding/prototext"
+	"github.com/google/go-tpm/tpm2"
+	"google.golang.org/protobuf/proto"	
+)
+var (
+	key      string
+	nonce    []byte
+	teenonce []byte
+	outputText string
+	gceZone  string
+    gceProjectID     string
+    gceProjectNumber uint64
+    gceInstanceName  string
+    gceInstanceID    uint64
+
+)
+
+var attestationKeys = map[string]map[tpm2.Algorithm]func (rw io.ReadWriter)(*client.Key,error)  {
+	"AK":{
+		tpm2.AlgRSA: client.AttestationKeyRSA,
+		tpm2.AlgECC: client.AttestationKeyECC,
+	},
+	"gceAK":{
+		tpm2.AlgRSA:client.GceAttestationKeyRSA,
+		tpm2.AlgECC:client.GceAttestationKeyECC,
+	},
+}
+
+var attestCmd = &cobra.Command{
+	Use:   "attest",
+	Short: "Create a remote attestation report",
+	Long: `Gather information for remote attestation.
+
+The Attestation report contains a quote on all available PCR banks ,a way to validate 
+the quote, and a TCG Event Log (Linux only).
+
+Use --key to specify the type of attestation key. It can be gceAK for GCE attestation
+key or AK for a custom attestation key.By default it uses gceAK.
+--algo flag override the public key algorithm for attestation key. If not provided then
+by default rsa is used.
+--nonce attaches even length extra data to the report and guarantees a fresh quote.
+--teenonce attaches a 64 bytes extra data to the attestation report of hardware and 
+guarantees a fresh quote.
+--output flag provides file name and location to store attestation report in the binary
+format. If filename is not provided then output will be printed on stdout.
+--outputText flag can be used to add text format of attestation report in a file.
+--gce-project-id and --gce-project-number used to verify attestation report.Either both
+will be used or neither.
+--gce-zone , --gce-instance-id and --gce-instance-name are optional flags.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		rwc,err := openTpm()
+		if(err != nil){
+			return err
+		}
+		defer rwc.Close()
+
+
+		createFunc := attestationKeys[key][keyAlgo]
+		
+		attestationKey , err := createFunc(rwc)
+		if err != nil {
+			return fmt.Errorf("failed to create attestation key: %v", err)
+		}
+		defer attestationKey.Close()
+
+
+		attestopts := client.AttestOpts{}
+		attestopts.Nonce = nonce
+		if(len(teenonce) != 0){
+			attestopts.TEENonce = teenonce
+			attestopts.TEEDevice,err = client.CreateSevSnpDevice()
```

Comment:
@deeglaze can you take a closer look at this section? We are tying it pretty closely with SEV-SNP for now, and I would prefer that our CLI commands work compatibly with future additions like TDX.